### PR TITLE
[#1639] va backtop, va switch, va color indicator nuxt fix

### PR DESCRIPTION
**Pull request by aluarius**
Related to #1639 (not closing, cause of item about docs, which is waiting for nuxt).

## Description
- Fixed error in Nuxt 3 with `VaColorIndicator` (`square` style disappears after page reload). It seems, that 0 as Number somehow rerenders as if it was Boolean. Changed to the String.
- Fixed hydration node mismatch with `VaSwitch` in Nuxt 3. Problem was solved by conditional rendering of labels, if there're no passed content slots or props for them.
- `VaBackTop`. Fixed Nuxt 3 error with undefined document (target is now binding after component was mounted). Default target was changed to `window` cause `document.body`, that was used before, always has `scrollTop` === 0. Window doesn't have scrollTop at all, so, in case of default target, will be used `scrollTo` method (and `scrollY` attribute for visibility behaviour as well) with smooth behivor (can't control speed, should mention this in docs).

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/components/va-backtop/VaBacktop.vue [hunk 1]
```diff
@@ -15,17 +15,16 @@
...
 
 export default defineComponent({
   name: 'VaBacktop',
   components: { VaButton },
   props: {
     target: {
-      type: [Object, String] as PropType<Element | string>,
-      default: () => window as any as Element,
+      type: [Object, String] as PropType<Element | string | null>,
+      default: null,
```
Comment (m0ksem) [inline packages/ui/src/components/va-backtop/VaBacktop.vue:27]:
```suggestion
      default: undefined,
```

File: packages/ui/src/components/va-backtop/VaBacktop.vue [hunk 2]
```diff
@@ -65,27 +71,41 @@ export default defineComponent({
...
+      const targetScrollValue = targetElement instanceof Window
+        ? targetElement.scrollY
+        : targetElement.scrollTop
+      visible.value = targetScrollValue > props.visibilityHeight
     }
 
     onMounted(() => {
       targetElement = getTargetElement()
-      targetElement?.addEventListener('scroll', handleScroll)
+      targetElement.addEventListener('scroll', handleScroll, true)
```
Comment (m0ksem) [inline packages/ui/src/components/va-backtop/VaBacktop.vue:105]: Does it prevent to capture scroll event from children?

File: packages/ui/src/components/va-backtop/VaBacktop.vue [hunk 3]
```diff
@@ -53,10 +52,17 @@ export default defineComponent({
...
+    let targetElement: Element | Window
+
+    const getTargetElement = () => {
+      if (props.target) {
+        return (typeof props.target === 'string'
+          ? document.querySelector(props.target)
+          : props.target) as Element
+      } else {
+        return window as Window
+      }
```
Comment (m0ksem) [inline packages/ui/src/components/va-backtop/VaBacktop.vue:64]:
```suggestion
    const getTargetElement = () => {
      if (!props.target) { return window }
      if (typeof props.target === 'string') { return document.querySelector(props.target) }
      
      return props.target
    }
```

**Review by m0ksem — APPROVED** [review]

File: packages/ui/src/components/va-backtop/VaBacktop.vue (hunk 2 above)
Comment (aluarius) [reply packages/ui/src/components/va-backtop/VaBacktop.vue:105]: One of the reasons, the second is that without capture scroll event listener not working properly for window.

**Pull request merged**